lantiq/xrx200: ImageBuilder: No rule to make target openwrt-lantiq-xrx200-arcadyan_arv7519rw22-squashfs-sysupgrade.bin

### Describe the bug

https://github.com/openwrt/docker/actions/runs/14251313226/job/39944529774#step:7:1224

```shell
[mktplinkfw2] firmware file "/builder/build_dir/target-mips_24kc_musl/linux-lantiq_xrx200/tmp/openwrt-lantiq-xrx200-tplink_vr200v-squashfs-sysupgrade.bin.new" completed
make[3]: *** No rule to make target '/builder/build_dir/target-mips_24kc_musl/linux-lantiq_xrx200/arcadyan_arv7519rw22-kernel.bin', needed by '/builder/build_dir/target-mips_24kc_musl/linux-lantiq_xrx200/tmp/openwrt-lantiq-xrx200-arcadyan_arv7519rw22-squashfs-sysupgrade.bin'.  Stop.
make[2]: *** [Makefile:263: build_image] Error 2
make[1]: *** [Makefile:159: _call_image] Error 2
make: *** [Makefile:332: image] Error 2
```

### OpenWrt version

Snapshot

### OpenWrt release

Snapshot

### OpenWrt target/subtarget

lantiq/xrx200

### Device

Default profile

### Image kind

Official downloaded image

### Steps to reproduce

_No response_

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

_No response_

### Diffconfig

```text

```

### Terms

- [x] I am reporting an issue for OpenWrt, not an unsupported fork.

> Well, its still failing on smoke-testing, now its missing:
> 
> ```
> make[3]: *** No rule to make target '/builder/build_dir/target-mips_24kc_musl/linux-lantiq_xrx200/arcadyan_arv7519rw22-kernel.bin', needed by '/
> ```

Grr, ok so when you do `make image` without an explicit `PROFILE`, the imagebuilder chooses `tplink_vr200` as its target, but somehow that implicit profile wants the `arcadyan_arv7519rw22` kernel?  Truly WTF.  

If I explicitly specify `make image PROFILE=tplink_vr200`, everything works fine.  I'm trying to figure out how those target makefiles are selecting their dependencies and mixing the profiles, it's somewhere inside wherever this ends up: https://github.com/openwrt/openwrt/blob/main/target/imagebuilder/files/Makefile#L264

Or maybe that makefile should just do a substitution when `PROFILE=<nothing>`, and pluck the first one from the profile list and make it explicit???

Or a dirtier workaround would be to change the docker smoke test to just use an explicit profile...

```
$ profile=$(awk '/PROFILE_NAME/ {sub(/DEVICE_/, "", $4); print $4}' .profiles.mk)
$ make info PROFILE=$profile
...
```

Right, it's internally "Default", but that makes a weird combo of the tplink and arcadyan profiles:
```
[mktplinkfw2] firmware file "build_dir/target-mips_24kc_musl/linux-lantiq_xrx200/tmp/openwrt-lantiq-xrx200-tplink_vr200v-squashfs-sysupgrade.bin.new" completed
make[3]: *** No rule to make target 'build_dir/target-mips_24kc_musl/linux-lantiq_xrx200/arcadyan_arv7519rw22-kernel.bin', needed by 'build_dir/target-mips_24kc_musl/linux-lantiq_xrx200/tmp/openwrt-lantiq-xrx200-arcadyan_arv7519rw22-squashfs-sysupgrade.bin'.  Stop.
```

So, how about I supplement my recent PR to also eliminate `Default` from `PROFILE_NAMES` in `.profiles.mk`?  That seems to work fine, `make image` then uses `DEVICE_avm_fritz3370-rev2-hynix` as the value for `USER_PROFILE` and it builds just fine, no other side effects seen with explicit targets (except of course, you can no longer say `make image PROFILE=Default`).

--- a/include/image.mk
+++ b/include/image.mk
@@ -567,6 +567,13 @@ endef
 
 define Device/Check/Common
   _PROFILE_SET = $$(strip $$(foreach profile,$$(PROFILES) DEVICE_$(1),$$(call DEVICE_CHECK_PROFILE,$$(profile))))
+  # Check if device is disabled and if so do not mark to be installed
+  ifeq ($$(DEFAULT),n)
+    _PROFILE_SET :=
+  endif
+  ifeq ($$(BROKEN),y)
+    _PROFILE_SET :=
+  endif
   DEVICE_PACKAGES += $$(call extra_packages,$$(DEVICE_PACKAGES))
   ifdef TARGET_PER_DEVICE_ROOTFS
     $$(eval $$(call merge_packages,_PACKAGES,$$(DEVICE_PACKAGES) $$(call DEVICE_EXTRA_PACKAGES,$(1))))

